lib/include/event-driven/vDrawSkin.h

```diff
@@ -0,0 +1,1159 @@
+        dX[ 1]=  385.83; dY[1]=-222.83;
+        dX[ 2]=128.62; dY[ 2]=-222.83;
+        dX[ 3]=0.0; dY[ 3]=0.0;
+        dX[ 4]=-128.62; dY[ 4]=-222.83;
+        dX[ 5]=-385.83; dY[ 5]=-222.83;
+        dX[ 7]=-257.2; dY[ 7]=0.0;
+    };
+
+    void initRepresentativeTaxels(std::vector<int> taxelMap){
+        //yarp::os::RecursiveLockGuard rlg(recursive_mutex);
+        std::list<int> mapp(taxelMap.begin(), taxelMap.end());
+        mapp.sort();
+        mapp.unique();
+        double s=0;
+
+        size_t mappsize = mapp.size();
+        for (size_t i = 0; i < mappsize; i++)
+        {
+            repr2TaxelList[mapp.front()]=vectorofIntEqualto(taxelMap,mapp.front());
+            mapp.pop_front();
+        }
+        //assign correct spacing between addresses: every 12 5 empty addresses
+        for (auto it =repr2TaxelList.begin();it!=repr2TaxelList.end();it++){
+            for (auto it2 =  it->second.begin(); it2 !=it->second.end(); it2++)
+            {
+                if(it->first==-1){
+                    s+=1;
+                    *it2 = 10000; //dummy value to avoid problems. The algorithm below works only for contiguous triangles in assigning a correct value
+                    //*it2 = (ceil(s/2)-1) *4;//heat taxels are reassigned to be within triangle (es. traingle ID 0--> 6 and 10, ID = 1 --> 22,26)
+                }
+                else{
+                    int triangleID = (it->first-3)/12;
+                    *it2 += triangleID *4;
+                }
+            }
+
+        }
+        // return true;
+    }
+
+
+
+
+    std::tuple<double,double> makeMap(int index)const{
+
+        // ID=(ID-3)/12;
+        int ID = getID(index)[0];
+        int taxelIntriangle =getID(index)[1];
+
+        std::vector<double> info = data.find(ID)->second;
+        const double scale=1/40.0;
+
+        double cx = info[0];
+        double cy = info[1];
+        double th = info[2];
+        double gain = info[3];
+        int lrMirror = int (info[4]);
+
+
+        const double CST=cos(DEG2RAD*(th+0));
+        const double SNT=sin(DEG2RAD*(th+0));
+
+        double x=scale*dX[taxelIntriangle];
+        double y=scale*dY[taxelIntriangle];
+        if (lrMirror==1) x=-x;
+
+        double u=cx+CST*x-SNT*y;
+        double v=cy+SNT*x+CST*y;
+        if ((taxelIntriangle!=6) && (taxelIntriangle!=10)) return std :: make_tuple(abs(u), v);
+        else return std :: make_tuple(-1000,-1000);//dummy location
+    };
+
+};
+
+
+
+class Right_hand : public PAD{
+    private:
+    int palmShift;
+//hand must be a wrapper for fingers and palm. Indeed the calibration file is unique
+    public:
+        Right_hand(int shift){
+            xoffset = 352;
+            yoffset = 400;
+            scaling =6;
+            palmShift = shift;
+            noise = 2300;
+            max_value = 10000;
+
+            dX = new double[48];
+            dY = new double[48];
+            std::cout<< "\nInitialise hand"<<std::endl;
```

Comment:
Again a lot of hardcoded values

Reply:
agreed. for now we will leave them as is